src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceHelpers.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.Extensions.Hosting.WindowsServices
     /// </summary>
     public static class WindowsServiceHelpers
     {
+        private static bool? _isWindowService;
```

Comment:
The field should be `readonly bool` so that JIT can turn it into constant in tiered code.